ding_spt_statistics.module

```diff
@@ -3,3 +3,153 @@
  * @file
  * Implementation of hooks and basic functionality goes here.
  */
+
+/**
+ * Implements hook_menu().
+ */
+function ding_spt_statistics_menu() {
+  $items = array();
+
+  $items['spt/redirect'] = array(
+    'title' => 'Ding SPT redirect',
+    'description' => 'Count click and redirect to the external resource',
+    'type' => MENU_CALLBACK,
+    'access arguments' => array('access content'),
+    'page callback' => 'ding_spt_statistics_redirect',
+  );
+  $items['spt/statistics'] = array(
+    'title' => 'Ding SPT statistics',
+    'description' => 'Count click and redirect to the external resource',
+    'type' => MENU_CALLBACK,
+    'access arguments' => array('view spt statistics'),
+    'page callback' => 'ding_spt_statistics_display',
+  );
+
+  return $items;
+}
+
+/**
+ * Implements hook_permission().
+ */
+function ding_spt_statistics_permission() {
+  return array(
+    'view spt statistics' => array(
+      'title' => t('View SPT statistics'),
+      'description' => t('View statistics for clicked external SPT resources.'),
+    ),
+  );
+}
+
+/**
+ * Count click and redirect to the resource.
+ */
+function ding_spt_statistics_redirect() {
+  // Register statistical data.
+  $hostname = !empty($_GET['hostname']) ? $_GET['hostname'] : FALSE;
+  ding_spt_statistics_register($hostname);
+
+  // Redirect to the external resource.
+  $path = !empty($_GET['path']) ? $_GET['path'] : FALSE;
+  drupal_goto($path);
+}
+
+/**
+ * Register hostname into the statistics table.
+ *
+ * @param $hostname string
+ *   Name of the external host.
+ */
+function ding_spt_statistics_register($hostname) {
+  if (!$hostname) {
+    return;
+  }
+  $result = db_select('ding_spt_statistics', 'dss')
+    ->fields('dss', array('hid', 'count'))
+    ->condition('hostname', $hostname)
+    ->execute()
+    ->fetchAssoc()
+  ;
```

Comment:
P.S 
for getting id - $lastid = db_query($query, $args, array('return' => Database::RETURN_INSERT_ID));